### DISCUSSION TITLE
5.0 Timeline / roadmap

### DISCUSSION THREAD

**Discussion by thec0keman** [openseadragon/openseadragon#2482] [General]
Is there a timeline or roadmap for releasing 5.0?

**Comment by iangilman** [comment]
Nothing specific. The most important thing is that we let https://github.com/openseadragon/openseadragon/pull/2310 bake for a while so we're confident it's solid. We're also interested in getting https://github.com/openseadragon/openseadragon/pull/2407 in for 5.0, but I suppose that depends on the timing.

At any rate, I expect it will be a while. Thank you for helping us find and sort out issues with the WebGL drawer... That helps get us closer!

**Comment by stefanheinze** [comment]
For our project we are developing at [ZEISS](https://www.zeiss.com) it would be very beneficial to have a kind of timeline for the next release.

We are using OSD in our project but had to remove the tiling support for our last product release, because of the problems we reported in #2481 . Adjusting the tile overlap was not an option because we have no control about the tile creation.
And as you can imagine the performance without the tiling is not acceptable for our users.

We are heading towards our next product release and need to know wether there is a chance to go with the OSD version 5.0 which includes the WebGL web renderer. From our first tests it seems it can solve our tiling problems.

Is it possible to get a pre-release version of OSD 5.0 with the WebGL implementation as npm package to easy integrate that version in our project and tests. As we are working with a lot of large images we can support by testing the WebGL implementation in a variaty of browsers to consider it is solid.
In case of errors it would be easier to refer to a specific pre-release version than testing with unversioned own build version.

**Reply by pearcetm** [reply]
> Is it possible to get a pre-release version of OSD 5.0 with the WebGL implementation as npm package to easy integrate that version in our project and tests.

You can install via npm directly from github like so: `npm install openseadragon/openseadragon` (with `--save` if you want, of course). This will install from the current master branch. I'd recommend installing from your own fork `npm install zeiss/openseadragon` (for example) for better control. See [this stackoverflow answer](https://stackoverflow.com/a/17509764/1214731) for syntax to do this from an enterprise github organization. Since you manage your own fork, you can give it whatever tags you want to keep track of the version at certain timepoints, and periodically pull from the upstream repo to stay up to date with new patches.

**Reply by pearcetm** [reply]
> As we are working with a lot of large images we can support by testing the WebGL implementation in a variaty of browsers to consider it is solid.

It would be great to have you test it out and report any bugs you may find so we can get it fixed up before the next release!
